arangod/Aql/Executor/CalculationExecutor.h

```diff
@@ -141,6 +140,9 @@ class CalculationExecutor {
   // Necessary for owned executors, which will not be exited when we call
   // exitContext; but only for assertions in maintainer mode.
   bool _hasEnteredExecutor;
+
+  // note: it is fine if this counter overflows
+  uint_fast16_t _killCheckCounter = 0;
```

Comment:
Shouldn't this be atomic since it is accessed from multiple threads?

Reply:
I guess it is not critical for the functionality, but ASAN will complain if indeed multiple threads access this in an uncontrolled way.

Reply:
I am not sure, but it is possible that since this is in an executor, the logic of AQL execution ensures that this particular member is only ever accessed by the thread which is currently executing the query snippet.

Reply:
The AQL executors are always accessed by a single thread only. There is no need to make this member thread-safe.

Reply:
Then all is good, thanks for the clarification!

Reply:
Thanks for answering